src/base/utils/misc.cpp

```diff
@@ -625,8 +625,17 @@ void Utils::Misc::openFolderSelect(const QString &absolutePath)
     if ((output == "dolphin.desktop") || (output == "org.kde.dolphin.desktop"))
         proc.startDetached("dolphin", QStringList() << "--select" << Utils::Fs::toNativePath(path));
     else if ((output == "nautilus.desktop") || (output == "org.gnome.Nautilus.desktop")
-             || (output == "nautilus-folder-handler.desktop"))
-        proc.startDetached("nautilus", QStringList() << "--no-desktop" << Utils::Fs::toNativePath(path));
+                 || (output == "nautilus-folder-handler.desktop")) {
+        proc.start("nautilus", {"--version"});
+        proc.waitForFinished();
+        QString nautilusVer = proc.readLine().simplified();
+        nautilusVer = nautilusVer.remove(QRegExp("[^0-9]")).left(3);
```

Comment:
remove all Char simbol
and Ver e.g. 3.27.1.2 (32712 > 3281, it's bad), then left first 3 simbol

Reply:
Why do you want to do this in such an unreliable way?
What about "3.2.9"? It will be treated as greater than "3.28.1" by your code.

Reply:
You should do it the way @Chocobo1 told you to. QByteArray is perfectly converted to QString.

Reply:
that about something like:
```
    proc.waitForFinished();
    int nautilusVer[3], i = 0;
    foreach(QString num, QStringList (QString (proc.readLine().simplified()).remove(QRegExp("[^0-9.]")).split(QLatin1Char('.')))) {
        nautilusVer[i] = num.toInt();
        cout << num.toInt() << endl;
        i++;
    }
    if ((nautilusVer[0] >= 3) | (nautilusVer[1] >= 28) | (nautilusVer[2] >= 0))

Reply:
>that about something like

No. Something like:
```c++
const QString nautilusVerStr = QString(proc.readLine().simplified()).remove(QRegExp("[^0-9.]");
using NautilusVersion = Utils::Version<int, 3>;
if (NautilusVersion::tryParse(nautilusVerStr, {1, 0, 0}) > NautilusVersion {3, 28}) {
    // ...
}
```